### PR TITLE
Site Transfers: Add handling for site ownership confirmation

### PR DESCRIPTION
**Pull request by wojtekn**
- Related to https://github.com/Automattic/dotcom-forge/issues/2563

## Proposed Changes

* Add a new screen to confirm the Site Transfer after clicking the link from the email.
* `/settings/start-site-transfer/${siteId}?site-transfer-confirm=${hash}`

## Testing Instructions

* Apply this diff if it's not deployed yet D112701-code
* Purchase a simple personal site
* Go to `/settings/general/${siteId}`
* Click on `Transfer site` and follow the steps.
* When you receive the email, copy the confirmation link and replace `wordpress.com` with the calypso live domain or `http://calypso.localhost:300` if you are running this branch from your computer.
* Observe a Loading bar and after a few seconds, you are redirected to `/sites`
* Access again to the already confirmed link and observe an error message.

## Screenshots

**Transferring**
Screenshot 2023-06-06 at 13 49 39

**Error**
Screenshot 2023-06-06 at 13 45 24

**Success**
- The success notice was introduced to `/sites` in this PR: https://github.com/Automattic/wp-calypso/pull/77660
image

## Screencast

https://github.com/Automattic/wp-calypso/assets/779993/02afc9c2-f154-4bd0-afe4-8c7d24520970

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by katinthehatsite** [comment]
@sejas is there something that you need to confirm from Reactor (Happiness team) in terms of the UX or something that needs clarification for you to get started?

**Comment by sejas** [comment]
I can start and suggest/implement a basic UI, and then we can improve it. Does it sound good?

**Comment by katinthehatsite** [comment]
Sounds good to me!

**Comment by sejas** [comment]
@katinthehatsite , @wojtekn , Can you give it a try?
There is a screencast with the whole flow.

**Comment by sejas** [comment]
In addition to improvements to the UI and copy changes, we could consider tracking errors.

File: client/my-sites/site-settings/site-owner-transfer/site-owner-transfer.tsx [hunk 1]
```diff
@@ -42,24 +39,16 @@ const SiteTransferComplete = () => {
 	);
 };
 
-const ActionPanelStyled = styled( ActionPanel )`
-	font-size: 14px;
-	font-weight: 400;
```
Comment (katinthehatsite) [inline client/my-sites/site-settings/site-owner-transfer/site-owner-transfer.tsx]: I was testing this and it seems that removing this bit affected the styles on "You have been sent..." screen for site transfer. This is what it looks in this PR:

Screenshot 2023-06-07 at 10 01 16 AM

This is what it should look like:

Screenshot 2023-06-07 at 10 05 04 AM

The site title and the text in the body should have the same font-size. It is a non-blocking feedback though 🚀

**Review by katinthehatsite — COMMENTED** [review]
This worked as expected for me with both Atomic and Simple sites. I only had one small note about the UI design change after this PR. Other than that, nice work!

**Comment by katinthehatsite** [comment]
One note that I had is that perhaps we could modify the error message from `The hash is incorrect` to something else that would be less technical for a regular user like `The confirmation link is incorrect`. I can already see the users reading that and being confused about what it means..

**Comment by kozer** [comment]
Nice work @sejas !
It works as expected!
I agree with the changes @katinthehatsite proposed.
One small note:
- When the user has the incorrect link, the bar appears for some time, before we display the error message. Can we change that and show nothing, or perhaps, a loading screen of some sort?

**Review by kozer — CHANGES_REQUESTED** [review]
Nice work @sejas !
It works as expected!
I agree with the changes @katinthehatsite proposed.
One small note:

When the user has the incorrect link, the bar appears for some time, before we display the error message. Can we change that and show nothing, or perhaps, a loading screen of some sort?

**Comment by sejas** [comment]
> When the user has the incorrect link, the bar appears for some time, before we display the error message. Can we change that and show nothing, or perhaps, a loading screen of some sort?

@kozer, I think the loading progress bar is already achieving that state of "loading".
That page makes only one request and it shows that loading bar until the request finishes.
If the request fails, the error appears. If the request succeeds the user navigates to `/sites`

**Comment by sejas** [comment]
> One note that I had is that perhaps we could modify the error message from `The hash is incorrect` to something else that would be less technical for a regular user like `The confirmation link is incorrect`. I can already see the users reading that and being confused about what it means..

@katinthehatsite , sure ! We can improve the copy.

The first part of the error is in the frontend, and the second part, `The hash is incorrect`, is generated in the backend and passed through the API. There are more possible errors.

fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sfvgr%2Qbjarefuvc%2Qgenafsre%2Spynff%2Qfvgr%2Qbjarefuvc%2Qgenafsre.cuc%3Se%3Qr2r2n47p%26zb%3Q4911%26sv%3Q159%23145%2Q180-og

243700374-4c93d041-0fc2-4cb4-aac6-94e9ff6a1bbf

@wojtekn, @katinthehatsite , anything against changing that copy in the backend?
We will need to create a separate diff for that.

Other alternatives would be:
- We show only a generic frontend error message for all possible errors.
- We return an error code from the backend and the frontend shows a different message for each error code.

File: client/my-sites/site-settings/site-owner-transfer/site-owner-transfer.tsx (hunk 1 above)
Comment (sejas) [reply client/my-sites/site-settings/site-owner-transfer/site-owner-transfer.tsx]: Thanks for pointing this out. I can see that behaviour on my screencast but I was not able to reproduce it.
After testing I see the font size of the title and paragraph to be 14px. Same as "production".


email-wpcalypso

**Comment by wojtekn** [comment]
> Other alternatives would be:
> We show only a generic frontend error message for all possible errors.

@sejas I'm curious if there are any benefits for showing a detailed error message here. I think that regardless of the message, users won't be able to do anything to fix it, and they will still need to check with support. Maybe we could say "There was an error confirming the site transfer. Please contact our support team for help."?

**Review by wojtekn — COMMENTED** [review]
Looks great and works well @sejas !

**Comment by sejas** [comment]
> Maybe we could say "There was an error confirming the site transfer. Please contact our support team for help."?

@wojtekn , done! Thanks for the suggestion.

https://github.com/Automattic/wp-calypso/pull/77798/commits/21d5119d6f789347d2e10581a566207d913c1ad7

File: client/my-sites/site-settings/site-owner-transfer/confirmation-transfer.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		confirmTransfer( confirmationHash );
+	}, [ confirmTransfer, confirmationHash ] );
+
+	if ( error ) {
+		return (
+			<Notice status="is-error" showDismiss={ false }>
+				<div data-testid="error">
+					<p>
+						{ translate(
+							'There was an error confirming the site transfer. Please contact our support team for help.'
```
Comment (katinthehatsite) [inline client/my-sites/site-settings/site-owner-transfer/confirmation-transfer.tsx]: I would maybe make this a link that goes to https://wordpress.com/help to make it easier for the user to find.

**Review by katinthehatsite — APPROVED** [review]
It works great!! I only had one small comment but it is non-blocking!

File: client/my-sites/site-settings/site-owner-transfer/confirmation-transfer.tsx (hunk 2 above)
Comment (sejas) [reply client/my-sites/site-settings/site-owner-transfer/confirmation-transfer.tsx]: Here is the link 👍 

https://github.com/Automattic/wp-calypso/pull/77798/commits/b12fbd9f1a24a986fcfefc52e920153d242c79b3

Screenshot 2023-06-14 at 11 10 13

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8045478

Thank you @wojtekn for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8045478).

**Pull request merged**